frontend/angular/style-guide.md

```diff
@@ -0,0 +1,972 @@
+│   │       ├── interfaces 
+│   │       │   └── specific.interface.ts
+│   │       ├── home.component.html
+│   │       ├── home.component.scss
+│   │       ├── home.component.spec.ts
+│   │       ├── home.component.ts
+│   │       ├── home-routing.module.ts
+│   │       └── home.module.ts
+│   └── services
+│       ├── api.service.ts
+│       └── local-store.service.ts
+├── app-routing.module.ts
+├── app.component.html
+├── app.component.scss
+├── app.component.spec.ts
+├── app.component.ts
+├── assets
+├── config
+├── environments
+└── styles 
+    ├── global.sccs
+    └── base
+        ├── buttons.sccs
+        └── forms.sccs
+```
+
+**Considerations:** 
+  - Make locating code intuitive, simple and fast
+  - Create a module and routing for each screen (for lazy routing)
+  - Each general component must have a module
+  - Do not repeat elements like interfaces, enumerations or functions if they are used more than once, include them in the helpers folder
+
+## Modules
+
+A module can be easily loaded in different places in your app. Also, the modules can be isolated for [testing](https://angular.io/guide/testing#angular-testbed) as a code unit.
+
+The module concept is based on some rules:
+
+- Small features.
+- Not injecting general services or external services.
+
+A module can contain a Component, Directive or Pipe. The key is: _If you need to reuse any of them, you need a shared module._
+
+### Module and Routing by screen
+
+A screen can contain some specific sections and components, so try to create module and routing.module by screen. Also remember in Angular the main concept is module.
+
+Module for Home screen, this is very important for [lazy loading](#lazy-loading) and [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns)
+
+```ts
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+
+import { HomeComponent } from './home.component';
+import { HomeRoutingModule } from './home-routing.module';
+
+@NgModule({
+  declarations: [ HomeComponent ],
+  imports: [
+    CommonModule,
+    HomeRoutingModule
+  ]
+})
+export class HomeModule {}
+```
+
+Routing module for Home screen
+```ts
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+
+import { HomeComponent } from './home.component';
+import { ResolverService } from './services/resolver.service'
+const routes: Routes = [
+  {
+    path: '',
+    component: HomeComponent,
+    children: [
+      {
+        path: '**',
+        redirectTo: ''
+      }
+    ]
+  }
+];
+
+@NgModule({
+  imports: [RouterModule.forChild(routes)],
+  exports: [RouterModule]
+})
+export class HomeRoutingModule { }
+
+```
+
+## Lazy Loading
+
+When your app is growing, it will be more lazy for a complete initial launch. Then, lazy loading will speeds up the app load time by splitting it into multiple bundles and loading them on demand. But, for this we need the Router help.
```

Comment:
```suggestion
When your app is growing, it will be more lazy for a complete initial launch. Then, lazy loading will speed up the app load time by splitting it into multiple bundles and loading them on demand. But, for this we need the Router help.
```